testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java

```diff
@@ -552,6 +552,44 @@ public void testJoinBaseTableWithView()
         assertAccessAllowed(viewOwnerSession, "DROP VIEW " + viewName);
     }
 
+    @Test
+    public void testFunctionOwners()
```

Comment:
This test is fine, but it could take some inspiration for the existing tests for views. For example, those tests are also checking things like: the access to the underlying resources (here: `my_test_function_inner`) is only checked on query/execute, not during creation. Though it could be different for functions.